pkg/sfu/downtrack.go

```diff
@@ -593,16 +596,17 @@ func (d *DownTrack) WriteRTP(extPkt *buffer.ExtPacket, layer int32) error {
 
 	d.streamAllocatorBytesCounter.Add(uint32(hdr.MarshalSize() + len(payload)))
 
-	if tp.isSwitchingToMaxLayer && d.onMaxSubscribedLayerChanged != nil && d.kind == webrtc.RTPCodecTypeVideo {
+	if tp.isSwitchingToMaxSpatial && d.onMaxSubscribedLayerChanged != nil && d.kind == webrtc.RTPCodecTypeVideo {
 		d.onMaxSubscribedLayerChanged(d, layer)
 	}
 
 	if extPkt.KeyFrame {
 		d.isNACKThrottled.Store(false)
 		d.rtpStats.UpdateKeyFrame(1)
 		d.logger.Debugw("forwarding key frame", "layer", layer)
+	}
 
-		// SVC-TODO - no need for key frame always when using SVC
+	if tp.isSwitchingToRequestSpatial {
```

Comment:
Introducing this. With SVC, the switch over does not need to happen on a key frame. So, a signal that current is switching to the key frame request layer is needed. Using this as signalled by selectors.